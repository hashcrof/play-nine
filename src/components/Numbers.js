import React, {Component} from 'react';
import {range} from 'lodash'

class Numbers extends Component {

  getNumberClass = (number) => {
    if(this.props.selectedNumbers.includes(number)){
      return 'selected';
    }
  }

  render() {
    return(
      <div className="card text-center">
        <div className="numbers">
          {Numbers.list.map(i =>
            <span key={i}
                  className={this.getNumberClass(i)}
                  onClick={() => this.props.onNumberClickHandler(i)}>
                  {i}
            </span>
          )}
        </div>
      </div>
    )
  }
}

Numbers.list = range(1,10); //TODO: what does this do?

export default Numbers;
