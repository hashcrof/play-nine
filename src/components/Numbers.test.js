import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Numbers from './Numbers'

const onNumberClickHandler = jest.fn()

it('renders the Numbers component', () =>{
  const output = shallow(
    <Numbers selectedNumbers={[2,4]} onNumberClickHandler={onNumberClickHandler}/>
  )
  expect(shallowToJson(output)).toMatchSnapshot()
});

it('renders nine numbers', () =>{
  const wrapper = mount(
    <Numbers selectedNumbers={[2,4]} onNumberClickHandler={onNumberClickHandler}/>
  )

  const div = wrapper.find('div.numbers')
  expect(div.props().children).toHaveLength(9)
});

it('calls onNumberClickHandler when a number is clicked', () =>{
  const wrapper = mount(
    <Numbers selectedNumbers={[2,4]} onNumberClickHandler={onNumberClickHandler}/>
  )

  const numberSpan = wrapper.find('span').first()
  const number = parseInt(numberSpan.text())
  numberSpan.simulate('click')
  expect(onNumberClickHandler).toBeCalledWith(number)
});
