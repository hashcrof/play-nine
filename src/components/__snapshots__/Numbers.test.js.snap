// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Numbers component 1`] = `
<div
  className="card text-center"
>
  <div
    className="numbers"
  >
    <span
      key="1"
      onClick={[Function]}
    >
      1
    </span>
    <span
      className="selected"
      key="2"
      onClick={[Function]}
    >
      2
    </span>
    <span
      key="3"
      onClick={[Function]}
    >
      3
    </span>
    <span
      className="selected"
      key="4"
      onClick={[Function]}
    >
      4
    </span>
    <span
      key="5"
      onClick={[Function]}
    >
      5
    </span>
    <span
      key="6"
      onClick={[Function]}
    >
      6
    </span>
    <span
      key="7"
      onClick={[Function]}
    >
      7
    </span>
    <span
      key="8"
      onClick={[Function]}
    >
      8
    </span>
    <span
      key="9"
      onClick={[Function]}
    >
      9
    </span>
  </div>
</div>
`;
